packages/wallets/phantom/src/adapter.ts

```diff
@@ -109,40 +136,54 @@ export class PhantomWalletAdapter extends BaseMessageSignerWalletAdapter {
         return this._readyState;
     }
 
-    async connect(): Promise<void> {
-        try {
-            if (this.connected || this.connecting) return;
-            if (this._readyState !== WalletReadyState.Installed) throw new WalletNotReadyError();
-
-            this._connecting = true;
-
-            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-            const wallet = window.phantom?.solana || window.solana!;
-
-            if (!wallet.isConnected) {
-                try {
-                    await wallet.connect();
-                } catch (error: any) {
-                    throw new WalletConnectionError(error?.message, error);
-                }
-            }
+    async connectWithWallet(): Promise<void> {
+        this._connecting = true;
 
-            if (!wallet.publicKey) throw new WalletAccountError();
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        const wallet = window.phantom?.solana || window.solana!;
 
-            let publicKey: PublicKey;
+        if (!wallet.isConnected) {
             try {
-                publicKey = new PublicKey(wallet.publicKey.toBytes());
+                await wallet.connect();
             } catch (error: any) {
-                throw new WalletPublicKeyError(error?.message, error);
+                throw new WalletConnectionError(error?.message, error);
             }
+        }
+
+        if (!wallet.publicKey) throw new WalletAccountError();
+
+        let publicKey: PublicKey;
+        try {
+            publicKey = new PublicKey(wallet.publicKey.toBytes());
+        } catch (error: any) {
+            throw new WalletPublicKeyError(error?.message, error);
+        }
+
+        wallet.on('disconnect', this._disconnected);
+        wallet.on('accountChanged', this._accountChanged);
+
+        this._wallet = wallet;
+        this._publicKey = publicKey;
 
-            wallet.on('disconnect', this._disconnected);
-            wallet.on('accountChanged', this._accountChanged);
+        this.emit('connect', publicKey);
+    }
 
-            this._wallet = wallet;
-            this._publicKey = publicKey;
+    async connect(): Promise<void> {
+        try {
+            if (this.connected || this.connecting) return;
 
-            this.emit('connect', publicKey);
+            if (this.readyState === WalletReadyState.Installed) {
+                await this.connectWithWallet();
+            } else if (this.readyState === WalletReadyState.Loadable) {
+                // redirect to the Phantom /browse universal link
+                // this will open the current URL in the Phantom in-wallet browser
+                const url = encodeURI(window.location.href);
+                const ref = encodeURI(window.location.origin);
+                const redirectUrl = `https://phantom.app/ul/browse/${url}?ref=${ref}`;
+                window.location.href = redirectUrl;
```

Comment:
If we take this approach, we might as well extract this logic to a private method also. Or since this is very little code, we could just handle this case first, and early return, and then inline the stuff that was extracted to `connectWithWallet`.